tests/mocha/procedure_map_test.js

```diff
@@ -36,20 +36,28 @@ suite.skip('Procedure Map', function() {
     });
 
     suite('procedure map updates', function() {
+      test('inserting a procedure does not trigger an update', function() {
+        const procedureModel =
+            new Blockly.procedures.ObservableProcedureModel(this.workspace);
+        this.procedureMap.set(procedureModel.getId(), procedureModel);
+
+        chai.assert.isFalse(
+            this.updateSpy.called, 'Expected no update to be triggered');
+      });
+
       test('adding a procedure does not trigger an update', function() {
-        this.procedureMap.addProcedure(
+        this.procedureMap.add(
             new Blockly.procedures.ObservableProcedureModel(this.workspace));
 
         chai.assert.isFalse(
             this.updateSpy.called, 'Expected no update to be triggered');
       });
 
       test('deleting a procedure triggers an update', function() {
-        const procedureModel =
-            new Blockly.procedures.ObservableProcedureModel(this.workspace);
-        this.procedureMap.addProcedure(procedureModel);
+        this.procedureMap.add(
+            new Blockly.procedures.ObservableProcedureModel(this.workspace));
 
-        this.procedureMap.deleteProcedure(procedureModel.getId());
+        this.procedureMap.delete([...this.procedureMap.values()][0].getId());
```

Comment:
why this change? the old way is more readable imo (here and below)

(i know the method name changed, but storing the procedureModel in a variable)

Reply:
I was moving the tests to actually use the fluent API hehe. I'll move things around so that we're storing the relevant bit in a variable though.

(side question: Do you think it'd be more readable as `this.procedureMap.getProcedures()[0].getId()`? )

Reply:
yeah, that would be more readable, but i think storing it in a variable is best.

you can have a separate test that tests the getProcedures function and one that tests the chain of the fluent api.

the justification is that if one of those methods break, you ideally want only the tests related to it to break, and not every single one of the tests, because that would slow down finding out the actual cause of the breakage. 

I think these are relevant snippets from go/readable-tests:
- "Create local variables when nontrivial values are referenced multiple times"
- "Write tests as straight-line code that doesn't force the reader to do mental computations to verify the test's correctness"
- "Write a separate test for each behavior defined by a method"
- "Write tests that call multiple methods on a class only when that sequence of calls defines a well-specified behavior"

Reply:
Do you think it's useful to test the chain of the fluent API? Imo I think that's well specified by the interfaces, and additional tests would just end up being change detectors. But idk, I'm still learning how to write tests well. (also acknowledging that you could have just been using that as an example haha)

I just wanted to change these so that they matched how we want people to actually use the API, not necessarily /test/ the API if that makes sense.

And wrt your snippets, were there any other changes you wanted me to make?

Reply:
> but in terms of how people would use it, i think it's important that you stick to public APIs, but not important that everything you do in the test would necessarily make sense to do in production, if that makes sense.

Sweet, totally makes sense!